File: backend/app/models/usuario.py
from dataclasses import dataclass
from typing import Optional


@dataclass
class Usuario:
    id: Optional[int]
    nome: str
    email: str
    telefone: str
    perfil: str          # 'CLIENTE' | 'TECNICO'
    especialidade: Optional[str]
    criado_em: Optional[str]

    def to_dict(self) -> dict:
        return {
            "id": self.id,
            "nome": self.nome,
            "email": self.email,
            "telefone": self.telefone,
            "perfil": self.perfil,
            "especialidade": self.especialidade,
            "criado_em": self.criado_em,
        }

File: mobile_cliente/lib/features/auth/models/usuario.dart
class Usuario {
  final int id;
  final String nome;
  final String email;
  final String telefone;
  final String perfil; // 'CLIENTE' | 'TECNICO'
  final String? especialidade;
  final String? criadoEm;

  const Usuario({
    required this.id,
    required this.nome,
    required this.email,
    required this.telefone,
    required this.perfil,
    this.especialidade,
    this.criadoEm,
  });

  factory Usuario.fromJson(Map<String, dynamic> json) => Usuario(
        id: json['id'] as int,
        nome: json['nome'] as String,
        email: json['email'] as String,
        telefone: json['telefone'] as String,
        perfil: json['perfil'] as String,
        especialidade: json['especialidade'] as String?,
        criadoEm: json['criado_em'] as String?,
      );
}
